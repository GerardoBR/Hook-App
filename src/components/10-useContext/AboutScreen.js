import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutScreen = () => {
    const {user, setsUser} = useContext(UserContext);
    const handlClik =()=>{
        setsUser({
            
        })
    }
    return (
        <div>
            <h1> About  Screen</h1>
            <hr/>
            <pre>
                {
                    JSON.stringify( user, null, 3 )
                }
            </pre>
            <button className = "btn btn-warning" onClick={ handlClik }> Logout</button>
        </div>
    )
}
