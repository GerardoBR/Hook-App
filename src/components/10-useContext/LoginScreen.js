import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
    // leer context 
    //  ocupar el setUser 
    const {setsUser, user} = useContext(UserContext)
    return (
        <div>
            <h1> Login Screen</h1>
            <hr/>
            <button className="btn btn-primary" onClick= {()=> setsUser({
                id: 123,
                name : 'Cesar'
            })}>Login</button>
        </div>
    )
}
