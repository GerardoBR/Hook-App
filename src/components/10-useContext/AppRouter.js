import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from "react-router-dom";
import { NavBar } from './NavBar';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
    return (
        <Router>
            <>
                <NavBar/>
                <div className= "container">

                    <Switch>
                        <Route
                            exact
                            path = "/"
                            component = { HomeScreen }
                        >
                        </Route>
                        <Route
                            path = "/about"
                            component = { AboutScreen }
                        >
                        </Route>
                        <Route
                            path = "/login"
                            component = { LoginScreen }
                        >
                        </Route>
                        {/* <Route component = { HomeScreen}/> */}
                        <Redirect to ="/"/>
                    </Switch>
                </div>

            </>
        </Router>
    )
}
