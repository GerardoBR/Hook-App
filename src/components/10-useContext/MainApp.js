import React, { useState } from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

export const MainApp = () => {
    const [user, setsUser] = useState({});

    return (
        <UserContext.Provider value = { {
            user,
            setsUser
        } }>
            <AppRouter/>
        </UserContext.Provider>
    )
}
