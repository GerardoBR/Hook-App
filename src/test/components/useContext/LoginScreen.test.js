import { mount, shallow } from "enzyme";
import { LoginScreen } from "../../../components/10-useContext/LoginScreen";
import { UserContext } from "../../../components/10-useContext/UserContext";

describe('Pruebas sobre el componetnte <LoginScreen/>', () => {
    const setsUser = jest.fn()

    const wrapper = mount(
        <UserContext.Provider value ={{
            setsUser
        }}>
            <LoginScreen/>
        </UserContext.Provider>
        
    );
    test('DEBE DE MOSTRARSE CORRECTAMENTE ', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    test('Debe de ejecutar el setUser con el argumento esperado', () => {
        
        wrapper.find('button').prop('onClick')();
        expect ( setsUser ).toHaveBeenCalledWith({
            id: 123,
            name : 'Cesar'
        })
    });
    
    
})
