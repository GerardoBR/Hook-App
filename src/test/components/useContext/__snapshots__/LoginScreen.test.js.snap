// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componetnte <LoginScreen/> DEBE DE MOSTRARSE CORRECTAMENTE  1`] = `
<div>
  <h1>
     Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
